docs/dev/metrics.md

```diff
@@ -0,0 +1,56 @@
+# CAO Metrics
+
+The Cluster Autoscaler Operator reports the following metrics:
+
+## Metrics provided by the controller runtime
+
+The [controller runtime](https://github.com/kubernetes-sigs/controller-runtime)
+integration with the operator provides several metrics about the internals of the
+operator. You can find more information about these metrics names and their labels through
+the following links:
+
+### Kubernetes controller metrics
+
+These metric names begin with `controller_runtime_reconcile_`. The labels
+`controller="cluster_autoscaler_controller"` and
+`controller="machine_autoscaler_controller"` can be used to refine queries against these metrics.
+* [Controller runtime reconciliation metrics implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/internal/controller/metrics/metrics.go)
+
+### Admission webhook metrics
+
+These metric names begin with `controller_runtime_webhook_`.  The label
+`webhook="/validate-clusterautoscalers"` can be used to refine the queries for these metrics.
+* [Controller runtime webhook metrics implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/webhook/internal/metrics/metrics.go)
+
+### API REST server metrics
+
+These metric names begin with `rest_client_` and `reflector_`. The labels
+`url` and `verb` can be used to refine these metrics further. The `url` label
+should contain the address to the Kubernetes API server, while `verb`
+containts the type of HTTP request (eg `"GET"`).
+* [Controller runtime Prometheus REST server metrics implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/metrics/client_go_adapter.go)
+
+### Prometheus work queue metrics
+
+These metric names begin with `workqueue_`. The labels
+`name="cluster_autoscaler_controller"` and `name="machine_autoscaler_controller"`
+can be used to refine queries against these metrics.
+* [Controller runtime Prometheus work queue metrics implementation](https://github.com/kubernetes-sigs/controller-runtime/blob/master/pkg/metrics/workqueue.go)
+
+## Metrics about the Prometheus collectors
+
+Prometheus provides some default metrics about the internal state
+of the running process and the metric collection. You can find more information
+about these metric names and their labels through the following links:
+
+* [Prometheus documentation, Standard and runtime collectors](https://prometheus.io/docs/instrumenting/writing_clientlibs/#standard-and-runtime-collectors)
+* [Prometheus client Go language collectors](https://github.com/prometheus/client_golang/blob/master/prometheus/go_collector.go)
+
+# Cluster Autoscaler Metrics
```

Comment:
Nit:  This seems like it could replace the first H1 at the top of this doc.

Reply:
this one is specifically about the autoscaler though, the top level link is about the operator. i was trying to draw a distinction, maybe there is a better way to specify?